surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/SmartTestingSurefireProvider.java

```diff
@@ -102,4 +107,27 @@ private File findFirstMatchingPom(File source) {
     private boolean isAnyDebugEnabled() {
         return bootParams.getMainCliOptions().contains(CommandLineOption.LOGGING_LEVEL_DEBUG) || configuration.isDebug();
     }
+
+    private Object getConsoleLogger() {
+        try {
+            Optional<Method> method = Arrays.stream(bootParams.getClass().getMethods())
+                .filter(this::isGetConsoleLoggerMethod)
+                .findFirst();
+            if (method.isPresent()) {
+                return method.get().invoke(bootParams);
+            }
+        } catch (IllegalAccessException | InvocationTargetException e) {
+            new DefaultLoggerFactory()
```

Comment:
I think you should throw an exception as well, because if not then this method will return null, which means that `consoleLogger` will be null and then when it is going to be called, we will get a `NPE`

Reply:
This should have been handled here: https://github.com/arquillian/smart-testing/pull/223/files#diff-72e56dc4b88e2415a66df5915b9e4b6cR40
there should be `if`

Reply:
The reason why I don't want to throw an exception is that incompatible logging doesn't have to mean that the whole ST tool won't work.

Reply:
> there should be if

fixed now